### DISCUSSION TITLE
mariadb and phpmyadmin

### DISCUSSION THREAD

**Discussion by sybanyt** [htpcBeginner/docker-traefik#269] [Q&A] [answered]
I am unable to login to phpmyadmin after following the guide. 
I have created a new user in mariadb and i can login with mysql when i docker exec into mariadb container:

Below is my docker compose file:
[docker-compose-t2.txt](https://github.com/htpcBeginner/docker-traefik/files/8599170/docker-compose-t2.txt)
 
When i attempt logging into myphpadmin page with the user I have created i get the error:
 mysqli::real_connect(): (HY000/2002): No such file or directory

**Comment by Maynardnaze** [comment]
just like this?
  # phpMyAdmin - Database management
  # Create a new user with admin privileges. Cannot login as MySQL root for some reason.
  phpmyadmin:
    <<: *common-keys-apps # See EXTENSION FIELDS at the top
    image: phpmyadmin/phpmyadmin:latest
    container_name: phpmyadmin
    networks:
      t2_proxy:
        ipv4_address: 192.168.90.17
    environment:
      - PMA_ABSOLUTE_URI=https://pma.change to domain name
      - PMA_HOST=mariadb
      - PMA_PORT=3306
      #- PMA_ARBITRARY=1
      - MYSQL_ROOT_PASSWORD_FILE=/run/secrets/mysql_root_password
    volumes:
      - $DOCKERDIR/appdata/pma:/etc/phpmyadmin/config.user.inc.php
    secrets:
      - mysql_root_password
    labels:
      - "traefik.enable=true"
      ## HTTP Routers
      - "traefik.http.routers.phpmyadmin-rtr.entrypoints=https"
      - "traefik.http.routers.phpmyadmin-rtr.rule=Host(`pma.$DOMAINNAME0`)"
      ## Middlewares
      - "traefik.http.routers.phpmyadmin-rtr.middlewares=chain-oauth@file"
      ## HTTP Services
      - "traefik.http.routers.phpmyadmin-rtr.service=phpmyadmin-svc"
      - "traefik.http.services.phpmyadmin-svc.loadbalancer.server.port=80"


and you have a proper user in .env right?
it clearly doesnt work with root

**Reply by Maynardnaze** [reply]
https://github.com/phpmyadmin/phpmyadmin/blob/master/config.sample.inc.php

**Reply by sybanyt** [reply]
I have been unable to get it to work after amending to what you have provided above.

 I am still getting the error after making modifications in my compose file and reviewing the link shared above:

 _Cannot log in to the MySQL server
mysqli::real_connect(): (HY000/2002): No such file or directory_

I had to amend this line:
- $DOCKERDIR/appdata/pma:/etc/phpmyadmin/config.user.inc.php
to 
- $DOCKERDIR/appdata/pma/config/user.inc.php:/etc/phpmyadmin/config.user.inc.php

I have created the config.user.inc.php file in the directory above with the contents from the linked github page before running docker-compose

Can you clarify for me your comment on having a proper user in .env.


After running docker compose, i did docker-exec into the mariadb container and i have created a user and granted all privileges on all databases with the command:

`create user 'username'@'%' identified by 'userpassword';
`
`grant all privileges on *.* to 'username'@'%';
`
`
flush privileges;
`
Attached is my docker-compose file
[docker-compose.txt](https://github.com/htpcBeginner/docker-traefik/files/8610095/docker-compose.txt)

**Comment by Maynardnaze** [comment]
And you cant get in with the created user through exec bash?

**Reply by sybanyt** [reply]
Through exec bash I am able to login but I can't login through the phpmyadmin webui

**Comment by sybanyt** [comment, marked as answer]
This was identified as a bug. Reference: https://github.com/phpmyadmin/docker/issues/361

Working for me now
